### ISSUE TITLE
auditable entity and field

### ISSUE THREAD

**Issue by vafabandi** [oroinc/platform#753]
hi everyone,
How to make an entity and fields auditable?
I write below annotions but in change history show nothing!
 *  "dataaudit"={
 *          "auditable"=true
 *      }
thanks
vafa

**Comment by zebimax** [comment]
Hello, you can read about data audit and how to work with it in the [documantation](https://www.orocrm.com/documentation/current/book/data/data-audits)

**Comment by vafabandi** [comment]
hi zebimax,
I do all all of work that said in documantation,  but it yet not work, change history every time is empty. please help me what I do.
![image](https://user-images.githubusercontent.com/31401448/31327883-d75a5e8c-acde-11e7-9733-a4c9ea2d48ca.png)

**Comment by ahmadgolzar** [comment]
Hi,
To give you more detail of what my colleague already explained about our problem. Here is how we have defined our entity:

```php
namespace Peymans\Bundle\PropertyBundle\Entity;

/**
 * Created by PhpStorm.
 * User: root
 * Date: 8/20/17
 * Time: 10:08 AM
 */
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;
use Doctrine\ORM\Mapping\JoinColumn;
use Doctrine\ORM\Mapping\JoinTable;
use Doctrine\ORM\Mapping\ManyToMany;
use Oro\Bundle\AttachmentBundle\Entity\Attachment;
use Oro\Bundle\EntityConfigBundle\Metadata\Annotation\Config;
use Oro\Bundle\EntityConfigBundle\Metadata\Annotation\ConfigField;
use Oro\Bundle\NoteBundle\Entity\Note;

/**
 * @ORM\Entity()
 * @ORM\Table(name="peymans_property")
 * @ORM\HasLifecycleCallbacks()
 * @Config(
 *  defaultValues={
 *      "entity"={
 *          "icon"="icon-tasks"
 *      },
 *      "ownership"={
 *          "owner_type"="USER",
 *          "owner_field_name"="owner",
 *          "owner_column_name"="user_owner_id",
 *          "organization_field_name"="organization",
 *          "organization_column_name"="organization_id"
 *      },
 *      "security"={
 *          "type"="ACL"
 *      },
 *      "dataaudit"={
 *          "auditable"=true
 *      }
 *
 *  }
 * )
 */
class Property
{

    /**
     * @var integer
     *
     * @ORM\Column(name="id", type="integer")
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="AUTO")
     */
    private $id;

    /**
     * @var string
     * @ORM\Column(name="name", type="string")
     * @ConfigField(
     *      defaultValues={
     *          "dataaudit"={
     *              "auditable"=true
     *          },
     *      }
     * )
     */
    private $name;


    /**
     * @return string
     */
    public function getName()
    {
        return $this->name;
    }

    /**
     * @param string $name
     */
    public function setName($name)
    {
        $this->name = $name;
    }
}
````

When we check entity manager UI, the entity is marked as auditable:
![entity](https://user-images.githubusercontent.com/8938399/31427917-d0c1adc6-ae75-11e7-924e-b2995ac60ca0.png)

but the name field is not:
![field](https://user-images.githubusercontent.com/8938399/31427940-e49a3278-ae75-11e7-8805-aba4326bd929.png)

What are we doing wrong. We'd appreciate it so much if you could help.

Thanks

**Comment by ahmadgolzar** [comment]
Does anyone have any idea what we are missing? any help would be very much appreciated.
Thanks

**Comment by digedag** [comment]
Hey @ahmadgolzar ,
your entity looks okay. Try to clear cache and execute this command: 
```bash
php app/console oro:entity-config:update --force --env=dev
```
If the field is still not auditable, you should look into database. All entity configurations for classes are persisted in `oro_entity_config`, all fields are in `oro_entity_config_field`. Find your attribute **name** in `oro_entity_config_field`. Check the base64 encoded column `data`. The **"auditable"=true** annotitation should be there.

**Comment by ahmadgolzar** [comment]
Hi @digedag 

Thank you so much for your reply. Your solution did the trick. 

We had already tried executing the `oro:entity-config:update` command, but without the `--env=dev` part. It did not have any effect. After executing the command with `--env=dev` part, the auditable attribute of the related field is set as true and audit is working properly. 

Of course we needed to make sure that `oro:message-queue:consume` was running as well so that audits were recorded in the database.

Thanks again
